### PR TITLE
MOBT-139: Ensure that wxmodal produces only day codes

### PR DESCRIPTION
**Pull request by bayliffe**
Ensure that wx-modal produces only day codes even for a single input file; prior to this change single files were returned unchanged, meaning night codes were returned in these cases alone.

This change also ensures that a `mode: time` cell method is added to give consistent metadata in the case of single file inputs.

Testing:
 - [x] Ran tests and they passed OK
 - [x] Added new tests for the new feature(s)

### CODE REVIEW

File: improver/wxcode/modal_code.py [hunk 1]
```diff
@@ -191,6 +196,8 @@ def process(self, cubes: CubeList) -> Cube:
         """
         # Handle case in which a single time is provided.
         if len(cubes) == 1:
```
Comment (MoseleyS) [inline improver/wxcode/modal_code.py:198]: I think that the special case is getting too special. It also leaves open the possibility that the time point on the input cube is not at the upper bound, which `_set_blended_times` would change. I've tested this alternative code (for the whole process method) and the unit and acceptance tests still pass.

```
        # Handle case in which a single time is provided.
        cube = MergeCubes()(cubes)
        self._unify_day_and_night(cube)

        if len(cube.coord("time").points) == 1:
            result = cube
            result.add_cell_method(self.mode_cell_method)
        else:
            result = cube.collapsed("time", self.aggregator_instance)
        self._set_blended_times(result)

        # Handle any unset points where it was hard to determine a suitable mode
        if (result.data == UNSET_CODE_INDICATOR).any():
            self._group_codes(result, cube)

        return result
```

File: improver_tests/wxcode/wxcode/test_ModalCode.py [hunk 2]
```diff
@@ -138,26 +138,34 @@ def test_expected_values(wxcode_series, expected):
...
+
+    A single data point is tested which means a single cube is passed in. This
+    ensures the metadata is consistent whether or not the input data has passed
+    through the modal aggregator."""
 
     def as_utc_timestamp(time):
         return timegm(time.utctimetuple())
 
     result = ModalWeatherCode()(wxcode_series)
+    print(result)
```
Comment (MoseleyS) [inline improver_tests/wxcode/wxcode/test_ModalCode.py:158]:
```suggestion
```

**Review by MoseleyS — CHANGES_REQUESTED** [review]

File: improver/wxcode/modal_code.py (hunk 1 above)
Comment (bayliffe) [reply improver/wxcode/modal_code.py]: Fine by me, it's just rearrangement. The time stuff should never be an issue unless something has gone wrong with the inputs, but this ensures against it I guess.

**Review by MoseleyS — APPROVED** [review]

**Review by gavinevans — CHANGES_REQUESTED** [review]
Thanks for these changes, @bayliffe 👍 

I think these changes are fine. The only change required is to modify the branch to point at the Hotfix_branch_1.0.2 branch. The Hotfix_branch_1.0.2 acceptance test branch should also be used.

**Comment by bayliffe** [comment]
Replaced by https://github.com/metoppv/improver/pull/1620 that goes into the hotfix branch. This will then be merged into master.

**Pull request closed without merging**